impeller/renderer/backend/gles/test/mock_gles.cc

```diff
@@ -0,0 +1,121 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <memory>
+
+#include "GLES3/gl3.h"
+#include "impeller/renderer/backend/gles/proc_table_gles.h"
+#include "impeller/renderer/backend/gles/test/mock_gles.h"
+
+namespace impeller {
+namespace testing {
+
+// OpenGLES is not thread safe.
+//
+// This mutex is used to ensure that only one test is using the mock at a time.
+static std::mutex g_test_lock;
+
+static std::weak_ptr<MockGLES> g_mock_gles;
+
+// Has friend visibility into MockGLES to record calls.
+void recordCall(const char* name) {
```

Comment:
```suggestion
void RecordGLCall(const char* name) {
```

It needs to be pascal case and `recordCall` is kind of a generic name to be placing in the testing namespace, so best to make it a bit more apropos.